src/mail/view/MailViewerToolbar.ts

```diff
@@ -152,9 +156,9 @@ export class MailViewerActions implements Component<MailViewerToolbarAttrs> {
 								"{current}": Math.round((operation.progress() / 100) * attrs.mails.length).toFixed(0),
 								"{total}": attrs.mails.length,
 							}),
-						exportMails(attrs.mails, locator.mailFacade, locator.entityClient, locator.fileController, operation.id, ac.signal).finally(
-							operation.done,
-						),
+						exportMails(attrs.mails, locator.mailFacade, locator.entityClient, locator.fileController, operation.id, ac.signal)
+							.then(this.handleExportEmailsResult)
```

Comment:
do not pass methods by reference, they lose `this` reference and it leads to *fun* errors